### PR TITLE
fix: update cpu feature flag detection & rust feature usage

### PR DESCRIPTION
**Pull request by cryptonemo**

### CODE REVIEW

**Comment by cryptonemo** [comment]
Hrm, interesting.  Never heard of `shellcheck`, but now I hate it ;-)

**Comment by cryptonemo** [comment]
@magik6k @Kubuxu: @ribasushi has been working on the CPU build issues where native builds can get illegal instructions.  This is due to detection issues, or people downloading optimized releases when they shouldn't be.  This PR fixes detection to use what the native cpu has instead of using 'no optimizations' or 'all optimizations'.

**Comment by cryptonemo** [comment]
Oh, this also updates the dependencies, including the proofs version, although that isn't necessarily related (the later version is for trusted setup verification differences, not specific to lotus)

**Comment by cryptonemo** [comment]
> Oh, this also updates the dependencies, including the proofs version, although that isn't necessarily related (the later version is for trusted setup verification differences, not specific to lotus)

If you need the Cargo.lock updates removed, feel free to edit them out, or let me know.

**Comment by cryptonemo** [comment]
Coincidentally also resolves #131 although I didn't see that issue until now (but realized it was an issue while working on this)

**Comment by cryptonemo** [comment]
If this is merged #130 can be closed.

File: install-filcrypto [hunk 1]
```diff
@@ -20,16 +20,16 @@ rust_sources_dir="rust"
 # building an optimized libfilcrypto (which takes advantage of some perf-boosting
 # instruction sets)
 #
-optimized_release_rustc_target_features=$(jq -r '.[].rustc_target_feature' < "${rust_sources_dir}/rustc-target-features-optimized.json")
+#optimized_release_rustc_target_features=$(jq -r '.[].rustc_target_feature' < "${rust_sources_dir}/rustc-target-features-optimized.json")
```
Comment (porcuquine) [inline install-filcrypto:24]: Should this just be deleted?

Comment (ribasushi) [reply install-filcrypto:24]: @porcuquine from elsewhere:
```
ribasushi:
@nemo a lot of commented-out stuff in this last commit
intended?

nemo:
shellcheck was telling me what wasn't used, so yes, I left some stuff for future reference
```

**Pull request merged**